[EDU-2250] BUILD MENU PT multiple origins (#340)
aziontech/docs: src/content/docs/pt-br/pages/build-journey/advanced-configurations/multiple-origins.mdx

---
title: Como configurar múltiplas origens com algoritmos de balanceamento de carga
description: Evite sobrecarga de servidor e congestionamento de tráfego entre origens, data centers ou provedores de nuvem através dos algoritmos de balanceamento de carga fornecidos pela Azion.
meta_tags: build
namespace: docs_guides_build_advanced_multi_origins

> permalink: /documentation/products/guides/build/multiple-origins/

Author: ```suggestion
permalink: /documentacao/produtos/guias/build/configure-multiplas-origens/
```

menu_namespace: buildMenu
---

> Quando você [cria uma edge application](/en/documentation/products/guides/start-with-a-template/), você precisa definir uma origem. No entanto, infraestruturas mais complexas e robustas podem precisar de múltiplas origens para a mesma aplicação. O módulo [Load Balancer](/en/documentation/products/edge-application/load-balancer/) permite que você use algoritmos de balanceamento de carga para gerenciar altos números de acesso e controlar os comportamentos de sua origem.

Reviewer: ```suggestion
Quando você [cria uma edge application](/pt-br/documentacao/produtos/comecar-com-um-template/), você precisa definir uma origem. No entanto, infraestruturas mais complexas e robustas podem precisar de múltiplas origens para a mesma aplicação. O módulo [Load Balancer](/en/documentation/products/edge-application/load-balancer/) permite que você use algoritmos de balanceamento de carga para gerenciar altos números de acesso e controlar os comportamentos de sua origem.
```

> :::caution[warning]

Reviewer: ```suggestion
:::caution[Atenção]
```

> Se o **Load Balancer** estiver ativado, a transferência de dados pode gerar custos relacionados ao uso. Verifique a [página de preços](https://www.azion.com/en/pricing/) para mais informações.

Reviewer: ```suggestion
Se o **Load Balancer** estiver ativado, a transferência de dados pode gerar custos relacionados ao uso. Verifique a [página de preços](https://www.azion.com/pt-br/precos/) para mais informações.
```

:::

> Os seguintes passos irão guiá-lo através de um cenário hipotético onde a default origin deve ser do tipo Load Balancer com o [algoritmo Round Robin](/en/documentation/products/edge-application/load-balancer/#balancing-method). Sinta-se à vontade para interpretar e modificar as instruções passo a passo de acordo com o seu caso de uso específico.

Reviewer: ```suggestion
Os seguintes passos irão guiá-lo através de um cenário hipotético onde a origem por default deve ser do tipo Load Balancer com o [algoritmo Round Robin](/pt-br/documentacao/produtos/edge-application/load-balancer/#método-de-balanceamento). Sinta-se à vontade para interpretar e modificar as instruções passo a passo de acordo com o seu caso de uso específico.
```

- A aplicação deve ter três origens.
  - O servidor primário deve ter uma maior capacidade de carga.
  - O servidor secundário deve ter uma capacidade de carga média, suficiente para lidar com grandes picos de tráfego de entrada.
  - O servidor de backup deve ter uma baixa capacidade de carga e só deve ser ativo em circunstâncias especiais.
- Cada um desses servidores deve ser hospedado em um provedor de armazenamento ou serviço de nuvem diferente, já que as falhas de servidor não ocorrem frequentemente ao mesmo tempo.
- Todas as origens devem conter o mesmo conteúdo e ser configuradas da mesma forma para a edge application.

| Origem | Função | Endereço | Capacidade de carga | Status |
| --- | --- | --- | --- | --- |
| 1 | Servidor primário | `example.com` | Alta capacidade de carga | Deve estar sempre ativo. |
| 2 | Servidor secundário | `example.net` | Capacidade de carga média | Deve estar sempre ativo. |
| 3 | Servidor de backup | `example.org` | Baixa capacidade de carga | Deve estar ativo apenas em caso de manutenção ou picos de tráfego. |

## Via RTM

> Primeiro, para habilitar o **Load Balancer** através do RTM:

Reviewer: ```suggestion
Para habilitar o **Load Balancer** através do RTM:
```

1. [Acesse o RTM](https://manager.azion.com).

> 2. No canto superior esquerdo da página, abra o **Products menu**, representado por três linhas horizontais, e então selecione **Edge Application**.

Reviewer: ```suggestion
2. No canto superior esquerdo da página, abra o **Products menu**, representado por três linhas horizontais, e selecione **Edge Application**.
```

3. Clique na edge application para a qual você quer configurar uma nova origem com balanceador de carga.
4. Na aba **Main Settings**, em **Modules**, ative o **Load Balancer**.
5. Clique no botão **Save**.

> :::caution[warning]

Reviewer: ```suggestion
:::caution[Atenção]
```

> Nesta seção, você modificará sua default origin, que é do tipo **Single Origin**, para uma origem do tipo **Load Balancer**. Editar sua default origin substituirá a origem que você configurou ao criar a edge application. Para criar uma origem diferente, veja [como configurar origens](/en/documentation/products/guides/build/work-with-origins/).

Reviewer: ```suggestion
Nesta seção, você modificará sua origem por default, que é do tipo **Single Origin**, para uma origem do tipo **Load Balancer**. Editar sua origem por default substituirá a origem que você configurou ao criar a edge application. Para criar uma origem diferente, veja [como configurar origens](/en/documentation/products/guides/build/work-with-origins/).
```

8. No campo **Server Role**, mantenha **Primary** selecionado.
9. Mantenha esta origem **Active**.
10. Adicione outra origem clicando no botão **+**.
11. No segundo campo **Address**, especifique o servidor secundário. Neste caso, `example.net`.
12. Devido à sua menor capacidade, no campo **Weight**, adicione um peso de `2` para esta origem.
13. No campo **Server Role**, selecione **Primary**. Esta origem tem um valor de peso menor, a origem preferida para conexões será a primeira.
14. Mantenha esta origem **Active**.
15. Adicione uma terceira origem clicando no botão **+**.
16. No campo **Address**, especifique a origem de backup. Neste caso, `example.org`.
17. Mantenha o campo **Weight** em branco. Ele assumirá um valor padrão de `1`.
18. No campo **Server Role**, selecione **Backup**.
19. Desative esta origem desligando o interruptor **Active**.
20. Clique no botão **Save**.

Aguarde alguns minutos para que as mudanças se propaguem. Em seguida, acesse sua aplicação ou use a linha de comando para fazer uma solicitação e verificar se as mudanças foram feitas.

## Via API

> 1. Execute a seguinte solicitação `GET` em seu terminal, substituindo `[TOKEN VALUE]` pelo seu [personal token](/en/documentation/products/guides/personal-tokens/) e a variável `<edge_application_id>` pelo [ID da sua edge application](/en/documentation/products/guides/build/configure-main-settings/) para recuperar a `<origin_key>` da sua origem padrão, que é um valor UUID:

Reviewer: ```suggestion
1. Execute a seguinte requisição `GET` em seu terminal, substituindo `[TOKEN VALUE]` pelo seu [personal token](/pt-br/documentacao/produtos/gestao-de-contas/personal-tokens/) e a variável `<edge_application_id>` pelo [ID da sua edge application](/pt-br/documentation/products/guides/build/definir-configuracoes-principais/) para recuperar a `<origin_key>` da sua origem padrão, que é um valor UUID:
```

```bash
curl --location 'https://api.azionapi.net/edge_applications/<edge_application_id> \
--header 'Accept: application/json; version=3' \
--header 'Authorization: Token [TOKEN VALUE]'
```

2. Você receberá uma resposta semelhante a esta:

```json
{
    "results": {
        "origin_id": <origin_id>,
        "origin_key": "<origin_key>",
        "name": "Default Origin",
        "origin_type": "single_origin",
        ...
    }
}
```

3. Copie o valor `<origin_key>`.

> 4. Execute a seguinte solicitação `PATCH` para modificar a origem padrão como segue, substituindo a chave e o id da aplicação na URL:

Reviewer: ```suggestion
4. Execute a seguinte requisição `PATCH` para modificar a origem padrão como segue, substituindo a chave e o id da aplicação na URL:
```

            "weight": 1,
            "server_role": "backup",
            "is_active": false
        }
    ],
    "host_header": "${host}",
    "method": "round_robin"
}'
```

5. Você receberá uma resposta semelhante a esta, confirmando que as mudanças foram feitas:

```json
{
    "results": {
        "origin_id": <origin_id>,
        "origin_key": "<origin_key>",
        "name": "Default Origin (RR3)",
        "origin_type": "load_balancer",
        "addresses": [
            {
                "address": "example.com",
                "weight": 3,
                "server_role": "primary",
                "is_active": true
            },
            {
                "address": "example.net",
                "weight": 2,
                "server_role": "primary",
                "is_active": true
            },
            {
                "address": "example.org",
                "weight": 1,
                "server_role": "primary",
                "is_active": false
            }
        ]
        "host_header": "${host}",
        "method": "round_robin"
        ...
    }
}
```

6. Aguarde alguns minutos para que a propagação ocorra.

:::tip

> Consulte a [documentação da API Azion](https://api.azion.com/) e a [especificação OpenAPI](https://github.com/aziontech/azionapi-openapi/) para saber mais sobre todas as funcionalidades disponíveis via API.

Reviewer: ```suggestion
Consulte a [documentação da API da Azion](https://api.azion.com/) e a [especificação OpenAPI](https://github.com/aziontech/azionapi-openapi/) para saber mais sobre todas as funcionalidades disponíveis via API.
```
